### PR TITLE
Display application line, address & network icon in Send token form

### PR DESCRIPTION
**Pull request by vyorkin**
## Summary

Enhances the "send token" form and "send transaction" screens by adding visual elements to improve UX and clarity.
Fixes missing application line in `SendTransaction`.

## Changes

* `ApplicationLine`: displays the application line for transactions
* Introduced `showApplicationLine` flag to control the display in send, swap, and transaction forms
* Adds a network icon to the recipient line, displayed conditionally

### CODE REVIEW

File: src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx [hunk 1]
```diff
@@ -1,22 +1,23 @@
-import React, { useLayoutEffect, useState } from 'react';
 import { animated, useTransition } from '@react-spring/web';
 import { capitalize } from 'capitalize-ts';
+import { ethers } from 'ethers';
```
Comment (vyorkin) [inline src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:3]: The only significant change here is the` import { ethers } from 'ethers';`. The other changes are related to using the "organize imports" code action (which, by the way, does a great job of organizing the imports)

File: src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx [hunk 2]
```diff
@@ -36,10 +38,11 @@ export function AddressActionDetails({
         <RecipientLine
           recipientAddress={recipientAddress}
           chain={chain}
+          showNetworkIcon={!showApplicationLine}
```
Comment (vyorkin) [inline src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:41]: We want to display network icon in recipient line only if we are not displaying application line

Comment (zerts) [reply src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:41]: Should it be `!(showApplicationLine && addressAction?.label)` then?

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx [hunk 3]
```diff
@@ -27,7 +32,24 @@ export function RecipientLine({
     <Surface style={{ borderRadius: 8, padding: '10px 12px' }}>
       <Media
         image={
-          <BlockieImg address={checksumAddress} size={36} borderRadius={6} />
+          <div
+            style={{
+              position: 'relative',
+              width: 36,
```
Comment (zerts) [inline src/ui/components/address-action/RecipientLine/RecipientLine.tsx:38]: I think, we can remove this size style. Inner element will handle it anyway.

File: src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx (hunk 2 above)
Comment (everdimension) [reply src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:41]: Seems so, though I would extract a helper variable

File: src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx (hunk 1 above)
Comment (everdimension) [reply src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:3]: Ideally library imports should be above project imports, but for some reason this doesn't work

File: src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx [hunk 4]
```diff
@@ -105,6 +106,23 @@ function ApplicationImage({
...
+  if (contractAddress) {
+    return contractAddress;
+  }
+
+  if (!action.label?.value) {
+    return null;
+  }
+
+  try {
+    return ethers.utils.getAddress(action.label?.value);
```
Comment (everdimension) [inline src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:120]: unnecessary `?.`?

Comment (everdimension) [inline src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:120]: you can use `toChecksumAddress.ts`, too

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx [hunk 5]
```diff
@@ -27,7 +32,24 @@ export function RecipientLine({
...
+              height: 36,
+            }}
+          >
+            <BlockieImg address={checksumAddress} size={36} borderRadius={6} />
+            {showNetworkIcon ? (
```
[the remark this answers is not in the document]
Comment (everdimension) [reply src/ui/components/address-action/RecipientLine/RecipientLine.tsx:43]: I think for nullish network the `NetworkIcon` will show a "???" fallback which is good to show (if we even reach this state)

**Review by everdimension — CHANGES_REQUESTED** [review]
All good, pls see these comments

https://github.com/zeriontech/zerion-wallet-extension/pull/550/files#r1636643902
https://github.com/zeriontech/zerion-wallet-extension/pull/550/files#r1636646636

File: src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx (hunk 1 above)
Comment (vyorkin) [reply src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:3]: it is not that smart yet to tell the difference between icon or library imports, updated

File: src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx (hunk 2 above)
Comment (vyorkin) [reply src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:41]: Yeah, that would be more correct, thank you ✅

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx (hunk 3 above)
Comment (vyorkin) [reply src/ui/components/address-action/RecipientLine/RecipientLine.tsx:38]: ✅

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (vyorkin) [reply src/ui/components/address-action/RecipientLine/RecipientLine.tsx:43]: Good point! I would always display the `NetworkIcon`, whether it's a fallback or not, to maintain consistency

File: src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx (hunk 4 above)
Comment (vyorkin) [reply src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:120]: thanks! refactoring leftover %) ✅

Comment (vyorkin) [reply src/ui/components/address-action/ApplicationLine/ApplicationLine.tsx:120]: 👍 , lets use `toChecksumAddress` here

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (zerts) [reply src/ui/components/address-action/RecipientLine/RecipientLine.tsx:37]: Will we remove condition from here?

**Review by zerts — APPROVED** [review]

File: src/ui/components/address-action/RecipientLine/RecipientLine.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (vyorkin) [reply src/ui/components/address-action/RecipientLine/RecipientLine.tsx:37]: I meant I would always display the `NetworkIcon` when the `RecipientLine` component has the `showNetworkIcon` flag enabled :) We don't need to display it when `showNetworkIcon === false` cause in other place we display the `NetworkIcon` on the `ApplicationLine` as well and we don't want the user to see 2 network icons at the same time

File: src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx [hunk 6]
```diff
@@ -28,18 +29,21 @@ export function AddressActionDetails({
...
         <RecipientLine
           recipientAddress={recipientAddress}
           chain={chain}
+          showNetworkIcon={showNetworkIcon}
           networks={networks}
         />
       ) : null}
-      {addressAction?.label && addressAction?.label.type !== 'to' ? (
+      {showApplicationLine && addressAction?.label ? (
         <ApplicationLine
```
Comment (everdimension) [inline src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:47]: I think this is all kinda backwards

It should be

```
const applicationLineVisible = Boolean(showApplicationLine && addressAction?.label);
...

              showNetworkIcon={!applicationLineVisible}
...

        {showApplicationLine && addressAction?.label ? ... : ...}
```

Comment (vyorkin) [reply src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx:47]: agree, your suggestion looks more readable, thank you!

**Review by everdimension — APPROVED** [review]

**Pull request merged**